### PR TITLE
added support for globally modifying rich editor toolbar buttons #2384

### PR DESCRIPTION
**Pull request by Harti**
This PR adds another tab to the administrator panel's editor settings: Toolbar Buttons.
You can configurate your global RichEditor toolbar buttons there.

If you've included the rich editor via a YAML file your toolbarButtons option will still override the global setting.

Please note that there is currently a CSS bug with horizontal rules not clearing floats in the Froala Editor.

### CODE REVIEW

**Comment by Harti** [comment]
this should also address #1743 and similar issues

**Comment by Harti** [comment]
@daftspunk 

(do you also require bounties for reviewing PRs?)

**Comment by LukeTowers** [comment]
This pull request has been made to the wrong branch. Please review [the contributing guidelines](https://github.com/octobercms/october/blob/master/CONTRIBUTING.md#pull-requests) as all PRs need to be made to the develop branch.

I'll fix it for you this time, but please ensure you make any future PRs to the develop branch, not the master branch.

@Harti we don't require bounties, but they help get an issue / pull request looked at faster. 
The support demand for October is increasing, while the available time the founders have to give remains the same. These issues will be handled in time, however there are ways you can help:

- Submit a pull request that introduces this feature and **wait for it to be reviewed**
- [Post a bug bounty](https://www.bountysource.com/teams/october) for this specific issue so that more people can become involved with reviewing it.
- [Make a financial contribution to the project](http://octobercms.com/fundraising) so we can allocate more time towards making improvements like these that benefit everyone

I appreciate your help in providing this change. I ask that you be patient with us though given the large demand for October support; it is purely based on volunteer time at the moment. If an issue is high priority for you; remember, the squeaky wheel gets the grease, so try contacting us on other platforms like IRC & Slack. Please just remember to be respectful of our time, and we'll usually be more than happy to help you out.

File: modules/backend/formwidgets/richeditor/partials/_richeditor.htm [hunk 1]
```diff
@@ -8,7 +8,7 @@
         <?php if ($fullPage): ?>data-fullpage="true"<?php endif ?>
         <?php if ($readOnly): ?>data-read-only="true"<?php endif ?>
         <?php if ($editorLang): ?>data-editor-lang="<?= $editorLang ?>"<?php endif ?>
-        <?php if ($toolbarButtons): ?>data-toolbar-buttons="<?= implode(',', $toolbarButtons) ?>"<?php endif ?>
+        <?php if ($toolbarButtons): ?>data-toolbar-buttons="<?= implode(',', $toolbarButtons) ?>"<?php elseif ($globalToolbarButtons): ?>data-toolbar-buttons="<?= str_replace(" ", "", $globalToolbarButtons) ?>"<?php endif ?>
```
Comment (LukeTowers) [inline modules/backend/formwidgets/richeditor/partials/_richeditor.htm]: Please place the `<?php elseif ($globalToolbarButtons):` on a new line for better readability

**Comment by LukeTowers** [comment]
Related: #2384, #2078, #1743 and https://github.com/rainlab/pages-plugin/issues/188. I'll review this right now.

File: modules/backend/lang/en/lang.php [hunk 2]
```diff
@@ -360,7 +360,9 @@
         'no_wrap' => 'Do not wrap tags',
         'no_wrap_comment' => 'The list of tags that should not be wrapped inside block tags.',
         'remove_tags' => 'Remove tags',
-        'remove_tags_comment' => 'The list of tags that are removed together with their content.'
+        'remove_tags_comment' => 'The list of tags that are removed together with their content.',
+        'toolbar_buttons' => 'Toolbar Buttons',
+        'toolbar_buttons_comment' => 'The Toolbar Buttons to be displayed in the Rich Editor. [fullscreen, bold, italic, underline, strikeThrough, subscript, superscript, fontFamily, fontSize, |, color, emoticons, inlineStyle, paragraphStyle, |, paragraphFormat, align, formatOL, formatUL, outdent, indent, quote, insertHR, -, insertLink, insertImage, insertVideo, insertAudio, insertFile, insertTable, undo, redo, clearFormatting, selectAll, html]',
```
Comment (LukeTowers) [inline modules/backend/lang/en/lang.php]: Please also change `The Toolbar Buttons to be displayed in the Rich Editor` to `The Toolbar Buttons to be displayed in the Rich Editor by default`

**Comment by LukeTowers** [comment]
Once those minor changes are made I'll merge this, thanks for your work!

**Comment by Harti** [comment]
@LukeTowers thank you for being patient with me and also fixing the PR to be on the correct branch. As you probably guessed I'm still a rookie in that field. On this note, I also probably shouldn't have used the default commit messages when editing in your review feedback (thanks for that, too!) as they're pretty nonexpressive.


Also, I am sorry for appearing disrespectful towards the project maintainer(s). I was just a little bummed as to how many PRs are lingering around here, partially without even having been reviewed. OctoberCMS is such a great software that was born and raised by such a passionate developer and an awesome contributor team, and it could become even greater. Obviously, peoples' lives and priorities change, some people get overwhelmed by an unexpected demand. 
While it is indeed not too much to ask for financial support for after having put tens of thousands of dollars worth into an open source project, I feel like copy-and-pasting that message in any bug report is a little out of place and disrespectful towards the community that *wants* to support you.

**Comment by carmeloMagna** [comment]
Hi @LukeTowers,

Thank you for this usefull update. Your last commit works very well.

I'm looking for a way to add a custom button in richeditor. Is possible to do that? And how? 
I followed some comments in #2078, but they did work.

Can you help me?

Thanks

**Comment by LukeTowers** [comment]
@Harti Believe me, it bums me out too. As much as I would love to be working on this project full time, that's just not financially viable for me to do so right now and still support myself. The point of that message is to let people know that I've seen their report and provide some options for them to get it addressed faster than it may otherwise be. If you have any suggestions for how I can communicate that in a way that comes off as less harsh, then I'd be welcome to hear them.

If you would like to help out, simply adding your input to issues or reviewing pull requests would be very helpful.

As far as the commit messages, don't worry too much about that; I usually just squash all the commits and merge the pull request in a single commit. Thanks again for taking care of this!

**Comment by LukeTowers** [comment]
@carmeloMagna Try asking on one of the following platforms for help with that: 
Forum: http://octobercms.com/forum
StackOverflow: http://stackoverflow.com/questions/tagged/octobercms
IRC: #october on Freenode or http://octobercms.com/chat
Slack: https://octobercms-slack.herokuapp.com/

**Comment by daftspunk** [comment]
Pull requests are always so tricky. Remember @Harti, it is a feature you want us to include in the project. It requires our attention and our time. Patience is an expectation of this arrangement, or we have given the route of including consideration, which is up to you. Making a contribution to the project is a great way to divert some of this attention your way. I'm not sure we should be recommending bounties anymore since we have the http://octobercms.com/fundraising page.

**Pull request merged**